pkg/cmd/server/origin/auth.go

```diff
@@ -355,8 +357,11 @@ func (c *AuthConfig) getAuthenticationHandler(mux cmdutil.Mux, errorHandler hand
 				}
 				passwordSuccessHandler := handlers.AuthenticationSuccessHandlers{c.SessionAuth, redirectSuccessHandler{}}
 
+				loginPath := path.Join(OpenShiftLoginPrefix, identityProvider.Name)
```

Comment:
The first loop can be super simple, just a check if we have more than one password+login provider